pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -555,31 +571,34 @@ func applyHTTPRouteDestination(
 		}
 	}
 
+	var hostname host.Name
 	if len(in.Route) == 1 {
-		processDestination(in.Route[0], serviceRegistry, listenerPort, hashByDestination, out, action)
+		hostname = processDestination(in.Route[0], serviceRegistry, listenerPort, hashByDestination, out, action)
```

Comment:
IIRC, it only returns a hostname in this branch, how do you build the stateful-session filter for the else branch

Reply:
Yes. I did only for non-weighted clusters. Will make change for weighted clusters now